### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by miheykrug**

### CODE REVIEW

File: lib/ext/gql/resolvers/proxy.ex [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
+      iex> Ext.Gql.Resolvers.Proxy.call([:extra, :test]).(%{extra: %{test: nil}}, %{}, %{})
+      {:ok, nil}
+
+      iex> Ext.Gql.Resolvers.Proxy.call(10).(%{spin: 11}, %{}, %{definition: %{schema_node: %{identifier: :spin}}})
+      {:ok, 11}
+
+      iex> Ext.Gql.Resolvers.Proxy.call(10).(%{spin: nil}, %{}, %{definition: %{schema_node: %{identifier: :spin}}})
+      {:ok, 10}
   """
   require IEx
```
Comment (atanych) [inline lib/ext/gql/resolvers/proxy.ex]: up

File: lib/ext/utils/base.ex [hunk 2]
```diff
@@ -102,16 +216,49 @@ defmodule Ext.Utils.Base do
...
+
+    ## Examples
+
+      iex> Ext.Utils.Base.to_keyword_list(%{first: "value1", second: "value2"})
+      [first: :value1, second: :value2]
+  """
+
   def to_keyword_list(data) do
-    Enum.reduce(data, [], fn {key, value}, acc -> [{to_atom(key), to_atom(value)} | acc] end)
+    Enum.reduce(data, [], fn {key, value}, acc -> acc ++ [{to_atom(key), to_atom(value)}] end)
```
Comment (atanych) [inline lib/ext/utils/base.ex]: I'm not sure this function is useful. Could we check we it's used?

`%{first: "value1", second: "value2"} |> Enum.to_list()` has almost the same functionality

File: lib/ext/utils/repo.ex [hunk 3]
```diff
@@ -1,4 +1,16 @@
 defmodule Ext.Utils.Repo do
+  @doc ~S"""
+  Return app name and repo
+
+  ## Examples
+
+      iex> Ext.Utils.Repo.get_config(Repo)
+      {:ext, Repo}
+
+      iex> Ext.Utils.Repo.get_config(nil)
```
Comment (atanych) [inline lib/ext/utils/repo.ex:10]: why get config for `nil` returned  `TestRepo`?

File: lib/mix/commands/deploy/send_slack_notification.ex [hunk 4]
```diff
@@ -12,8 +12,17 @@ defmodule Mix.Commands.Deploy.SendSlackNotification do
   end
 
   def call(%{tag: tag, env_name: env_name} = context, type) do
-    message = messages(env_name, tag)[type]
-    Ext.Commands.SendToSlack.call(Helper.settings()[:slack_token], Helper.settings()[:slack_channel], message)
+    payload = %Ext.Sdk.Request{
+      options: %{
+        url_params: %{
+          token: Helper.settings()[:slack_token],
```
Comment (atanych) [inline lib/mix/commands/deploy/send_slack_notification.ex]: why we can not use `GET params` instead of passing directly to url?

File: config/config.exs [hunk 5]
```diff
@@ -33,3 +33,8 @@ use Mix.Config
 config :logger, :console,
   format: "$date $time $metadata[$level] $message\n",
   metadata: [:sdk, :method, :url]
+
+case Mix.env() do
+  :test -> import_config "test.exs"
```
Comment (atanych) [inline config/config.exs]: "#{Mix.env()}"

**Review by atanych — APPROVED** [review]
